music_assistant/server/providers/snapcast/__init__.py

```diff
@@ -691,7 +678,14 @@ async def _stop_builtin_server(self) -> None:
     async def _start_builtin_server(self) -> None:
         """Start the built-in Snapserver."""
         if self._use_builtin_server:
-            self._builtin_server_retry = 0
             self._snapserver_started = asyncio.Event()
             self._snapserver_runner = asyncio.create_task(self._builtin_server_runner())
             await asyncio.wait_for(self._snapserver_started.wait(), 10)
+
+    def _handle_disconnect(self, exc: Exception) -> None:
+        """Handle disconnect callback from snapserver."""
+        self.logger.info(
+            "Connection to SnapServer lost, reason: %s. Reloading provider in 5 seconds.", str(exc)
+        )
+        # schedule a reload of the provider
+        self.mass.call_later(5, self.mass.config.reload_provider(self.instance_id))
```

Comment:
Bit of an edge case, but if the provider is removed while the timer is waiting, reload_provider explodes. (It calls get_provider_config, which will throw a KeyError if the provider was removed).

I think maybe just make it fail silently?:

```
class ConfigController:
    ...
    @api_command("config/providers/reload")
    async def reload_provider(self, instance_id: str) -> None:
        """Reload provider."""
        try:
            config = await self.get_provider_config(instance_id)
        except KeyError:
            # Provider was removed before we could reload it
            return
        await self._load_provider_config(config)```

?

Reply:
yeah - sounds reasonable